src/resolve/resolved-dns-transaction.c

```diff
@@ -1962,11 +1962,13 @@ int dns_transaction_go(DnsTransaction *t) {
                         assert_not_reached("bad protocol");
                 }
 
-                r = sd_event_add_time(
+                assert(!t->timeout_event_source);
+
+                r = sd_event_add_time_relative(
                                 t->scope->manager->event,
                                 &t->timeout_event_source,
                                 clock_boottime_or_monotonic(),
-                                ts + jitter, accuracy,
+                                jitter, accuracy,
```

Comment:
This moves the next base of next attempt from sd_event_now(), i.e. from the most recent event loop iteration beginning, to "now". I guess that's appropriate.

Reply:
actually `ts` was initialized to sd_event_now() too, so this isn't changed. Besides some more careful overflow checking this code shouldn't be altered at all i think by this commit